develop/tutorials/articles/360-osgi-basics-for-liferay-development/03-importing-packages.markdown

```diff
@@ -86,26 +86,50 @@ detects their use in the WAR's JSPs, descriptor files, and classes (in
 descriptor files. It adds package imports for classes that are neither found in
 the plugin's `WEB-INF/classes` folder nor in embedded JARs. 
 
-Packages such as Java platform packages are not versioned. OSGi Portable Java Contract Definitions let you get Java packages based on the JSR specification you're using. 
+Java platform packages (e.g., `javax.portlet.*` ) are handled differently.  
 
-## Using Portable Java Contracts [](id=using-portable-java-contracts)
+## Importing Java Platform Packages [](id=importing-java-platform-packages)
 
-@product@'s OSGi framework wires your module with Java platform packages based
-on the JSR specification of the Java API artifact you use and the Java platform
-packages available at run time. Here's how it works for different tool chains:
+You'll want to make sure the Java platform APIs you compile against are
+available at runtime. Since the API packages aren't versioned, you must use OSGi
+Portable Java Contracts to assure Java packages for the JSR you require are
+provided at runtime. 
 
--   Liferay's Gradle defaults in Blade CLI and Liferay @ide@ projects inherit 
-    the JavaPortlet contract, which generates Java platform package imports
-    automatically. 
--   If you use any other tool chain that provides bnd, add the
-    `-contract: JavaPortlet` to your bnd instructions. 
--   If you have neither the Liferay Gradle defaults or bnd, add the Java   
-    platform package imports (e.g., `Import-Package: javax.portlet`) to your
-    OSGi manifest, but DO NOT specify any package version--let the OSGi 
-    framework wire your module to an appropriate provider.  
+Here's how to set up Java platform API contracts in different module projects:
 
-For Portable Java Contract Definition details, see 
-[https://www.osgi.org/portable-java-contract-definitions/](https://www.osgi.org/portable-java-contract-definitions/). 
+- **Blade CLI and Liferay @ide@ projects** handle it automatically! They set up
+Java contracts based on the versions of `JavaPortlet` and `JavaServlet` you
+compiled against. 
+
+    - **Gradle projects** add the `JavaPortlet` and `JavaPortlet` contracts to
+    the module JAR manifest on assembly.
+    - **Maven projects** have `-contract` instructions for `JavaPortlet` and
+    `JavaPortlet` in their `bnd.bnd` file. 
+
+- **Other projects that use bnd** that are not created using Blade CLI or
+Liferay @ide@ must specify contracts in their `bnd.bnd` file. The Portable Java
+Contracts framework adds `Require-Capability` entries that contract the Java
+APIs and *removes* version information from the corresponding Java API
+`Import-Package` entries.
+
+        -contract: JavaPortlet
+        -contract: JavaServlet 
```

Comment:
I don't think you should put these on separate instructions because people will think it's correct, when the second will invalidate the first. I think you should use: `contract: JavaPortlet, JavaServlet`